src/transport/SecureSessionMgr.cpp

```diff
@@ -309,10 +311,10 @@ void SecureSessionMgrBase::HandleDataReceived(const PacketHeader & packetHeader,
 #if CHIP_SYSTEM_CONFIG_USE_LWIP
         /* This is a workaround for the case where PacketBuffer payload is not
            allocated as an inline buffer to PacketBuffer structure */
-        origMsg = msg;
-        msg     = PacketBuffer::NewWithAvailableSize(len);
-        VerifyOrExit(msg != nullptr, ChipLogError(Inet, "Insufficient memory for packet buffer."));
-        msg->SetDataLength(len, msg);
+        origMsg.Adopt(msg.Release());
```

Comment:
Again, `operator=` taking rvalue ref seems better to me....

Reply:
We can do these as followups. I believe this is just as easy to read as `origMsg = std::move(msg)` and does the same thing.

Reply:
Followup works for me.